frontend/app/.server/domain/multi-channel/case-api-service-default.ts

```diff
@@ -6,18 +6,45 @@ import { ErrorCodes } from '~/errors/error-codes';
 
 export function getDefaultSinCaseService(): SinCaseService {
   return {
-    getSinCases: async (): Promise<SinCaseDto[]> => {
+    listAllSinCases: async (): Promise<SinCaseDto[]> => {
       const response = await fetch(`${serverEnvironment.INTEROP_SIN_REG_API_BASE_URL}/cases`);
-      if (response.status === 404) return [];
-      if (!response.ok) throw new AppError(`Cases not found.`, ErrorCodes.SIN_CASE_NOT_FOUND);
-      return response.json();
+
+      if (!response.ok) {
+        const errorMessage = `Failed to retrieve all SIN cases. Server responded with status ${response.status}.`;
+        throw new AppError(errorMessage, ErrorCodes.XAPI_API_ERROR);
+      }
+
+      return await response.json();
     },
 
-    getSinCaseById: async (id: string): Promise<SinCaseDto | undefined> => {
+    findSinCaseById: async (id: string): Promise<SinCaseDto | undefined> => {
       const response = await fetch(`${serverEnvironment.INTEROP_SIN_REG_API_BASE_URL}/cases/${id}`);
-      if (response.status === 404) return undefined;
-      if (!response.ok) throw new AppError(`Case with ID '${id}' not found.`, ErrorCodes.SIN_CASE_NOT_FOUND);
-      return response.json();
+
+      if (response.status === 404) {
```

Comment:
For better readability, consider using `HttpStatusCodes.NOT_FOUND` everywhere.

Reply:
I didn't that existed. 👍 
Fixed.